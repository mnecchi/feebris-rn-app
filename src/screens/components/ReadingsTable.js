import React from 'react';
import {Table, Row, Rows} from 'react-native-table-component';

const ReadingsTable = ({readings, tableStyle, cellStyle, headerStyle}) => {
  if (!Array.isArray(readings) || readings.length === 0) {
    return null;
  }

  const data = readings.reduce((acc, reading) => {
    acc.push([
      new Date(reading.createdAt).toLocaleString(),
      reading.temperature,
      reading.cough ? 'ğŸ¤§' : '',
      reading.feverInLast5Days ? 'ğŸ¤’' : '',
      reading.isFlu ? 'ğŸ’Š' : '',
    ]);

    return acc;
  }, []);

  return (
    <Table borderStyle={tableStyle}>
      <Row
        style={headerStyle}
        data={['date', 'temp.', 'cough', '5 days fever', 'flu']}
        flexArr={[2, 1, 1, 1, 1]}
        textStyle={cellStyle}
      />
      <Rows textStyle={cellStyle} data={data} flexArr={[2, 1, 1, 1, 1]} />
    </Table>
  );
};

export default ReadingsTable;
