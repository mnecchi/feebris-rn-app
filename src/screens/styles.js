import {StyleSheet} from 'react-native';

// Common Styles

const labelText = {fontSize: 30, fontWeight: 'bold', color: '#444'};
const fieldContainer = {
  justifyContent: 'center',
  alignItems: 'center',
  margin: 16,
  maxWidth: '80%',
};
const switchContainer = {
  ...fieldContainer,
  textAlignVertical: 'center',
};

const iconInputContainer = {
  flexDirection: 'row',
  marginTop: 8,
  alignItems: 'center',
  justifyContent: 'center',
};

const buttonContainer = {
  padding: 8,
  backgroundColor: 'rgba(60, 142, 183, 1)',
  borderRadius: 8,
  marginTop: 64,
  width: '80%',
  maxWidth: 600,
  justifyContent: 'center',
  alignItems: 'center',
};

const buttonText = {...labelText, color: 'white'};

/**
 * Temperature, cough and fever screens styles
 */
const homeStyle = StyleSheet.create({
  container: {
    flex: 1,
    height: '100%',
    alignItems: 'center',
  },
  // TEMPERATURE
  temperatureContainer: fieldContainer,
  temperatureText: labelText,
  temperatureInput: {
    borderColor: '#ccc',
    borderWidth: 1,
    borderRadius: 8,
    padding: 8,
    fontSize: 20,
    width: 100,
    margin: 8,
  },
  temperatureInputError: {
    borderColor: 'red',
  },
  temperatureInputContainer: iconInputContainer,
  // COUGH
  coughContainer: switchContainer,
  coughText: {...labelText, paddingRight: 8},
  coughSwitchContainer: iconInputContainer,
  // FEVER IN LAST 5 DAYS
  feverContainer: switchContainer,
  feverText: {...labelText, paddingRight: 8},
  feverSwitchContainer: iconInputContainer,
  // BUTTON
  buttonContainer,
  buttonText,
});

/**
 * Results screen styles
 */
const resultsStyle = StyleSheet.create({
  container: {flex: 1, flexDirection: 'column'},
  topContainer: {
    alignItems: 'center',
    paddingBottom: 32,
    borderColor: '#ccc',
    borderWidth: 1,
  },
  // FLU CHECK RESULT
  fluResultContainer: {
    alignItems: 'center',
    marginTop: 32,
    marginHorizontal: 16,
  },
  fluResultText: {
    ...labelText,
  },
  fluResultFlu: {
    color: 'red',
  },
  fluResultNoFlu: {
    color: 'green',
  },
  // RESULTS TABLE
  tableContainer: {
    marginTop: 16,
    marginHorizontal: 8,
  },
  readingsTable: {
    borderWidth: 1,
    borderColor: '#ccc',
  },
  readingsTableHeader: {
    backgroundColor: 'rgba(60, 142, 183, 0.5)',
  },
  readingTableCell: {
    margin: 8,
    textAlign: 'center',
  },
  // BUTTON
  buttonContainer,
  buttonText,
});

export {homeStyle, resultsStyle};
