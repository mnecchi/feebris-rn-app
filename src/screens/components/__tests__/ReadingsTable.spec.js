import React from 'react';
import 'react-native';
import ReadingsTable from '../ReadingsTable';
import renderer from 'react-test-renderer';
import {resultsStyle} from '../../styles';

describe('should render successfully', () => {
  it('should return null if no array or empty array is passed', () => {
    const tree = renderer
      .create(
        <ReadingsTable
          readings={null}
          headerStyle={resultsStyle.readingsTableHeader}
          cellStyle={resultsStyle.readingTableCell}
          tableStyle={resultsStyle.readingsTable}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display a table with results', () => {
    const readings = [
      {
        temperature: 39,
        cough: true,
        feverInLast5Days: true,
        isFlu: true,
      },
      {
        temperature: 36,
        cough: false,
        feverInLast5Days: false,
        isFlu: false,
      },
    ];

    const tree = renderer
      .create(
        <ReadingsTable
          readings={readings}
          headerStyle={resultsStyle.readingsTableHeader}
          cellStyle={resultsStyle.readingTableCell}
          tableStyle={resultsStyle.readingsTable}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
