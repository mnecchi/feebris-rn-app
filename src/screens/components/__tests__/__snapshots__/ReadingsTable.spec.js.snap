// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render successfully should display a table with results 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#ccc",
        "borderLeftWidth": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        0,
        Object {
          "flexDirection": "row",
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "rgba(60, 142, 183, 0.5)",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderColor": "#ccc",
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          Object {
            "justifyContent": "center",
          },
          undefined,
          undefined,
          Object {
            "flex": 2,
          },
          false,
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        date
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#ccc",
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          Object {
            "justifyContent": "center",
          },
          undefined,
          undefined,
          Object {
            "flex": 1,
          },
          false,
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        temp.
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#ccc",
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          Object {
            "justifyContent": "center",
          },
          undefined,
          undefined,
          Object {
            "flex": 1,
          },
          false,
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        cough
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#ccc",
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          Object {
            "justifyContent": "center",
          },
          undefined,
          undefined,
          Object {
            "flex": 1,
          },
          false,
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        5 days fever
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#ccc",
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          Object {
            "justifyContent": "center",
          },
          undefined,
          undefined,
          Object {
            "flex": 1,
          },
          false,
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        is flu
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 6,
        },
        0,
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          0,
          Object {
            "flexDirection": "row",
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderColor": "#ccc",
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            Object {
              "justifyContent": "center",
            },
            undefined,
            undefined,
            Object {
              "flex": 2,
            },
            false,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          Invalid Date
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderColor": "#ccc",
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            Object {
              "justifyContent": "center",
            },
            undefined,
            undefined,
            Object {
              "flex": 1,
            },
            false,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          39
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderColor": "#ccc",
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            Object {
              "justifyContent": "center",
            },
            undefined,
            undefined,
            Object {
              "flex": 1,
            },
            false,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          ðŸ¤§
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderColor": "#ccc",
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            Object {
              "justifyContent": "center",
            },
            undefined,
            undefined,
            Object {
              "flex": 1,
            },
            false,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          ðŸ¤’
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderColor": "#ccc",
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            Object {
              "justifyContent": "center",
            },
            undefined,
            undefined,
            Object {
              "flex": 1,
            },
            false,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          ðŸ’Š
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          undefined,
          0,
          Object {
            "flexDirection": "row",
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderColor": "#ccc",
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            Object {
              "justifyContent": "center",
            },
            undefined,
            undefined,
            Object {
              "flex": 2,
            },
            false,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          Invalid Date
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderColor": "#ccc",
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            Object {
              "justifyContent": "center",
            },
            undefined,
            undefined,
            Object {
              "flex": 1,
            },
            false,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          36
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderColor": "#ccc",
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            Object {
              "justifyContent": "center",
            },
            undefined,
            undefined,
            Object {
              "flex": 1,
            },
            false,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderColor": "#ccc",
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            Object {
              "justifyContent": "center",
            },
            undefined,
            undefined,
            Object {
              "flex": 1,
            },
            false,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderColor": "#ccc",
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            Object {
              "justifyContent": "center",
            },
            undefined,
            undefined,
            Object {
              "flex": 1,
            },
            false,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`should render successfully should return null if no array or empty array is passed 1`] = `null`;
